### PR TITLE
[12.0][ADD] purchase_deposit

### PR DESCRIPTION
**Pull request by kittiu**
Done
- [X] code review
- [X] readme files
- [X] test script

This module add ability to create purchase deposit, similar to sales deposit.

### CODE REVIEW

**Comment by kittiu** [comment]
Currency there is following error,
```
************* Module purchase_deposit
purchase_deposit/__init__.py:1: [C7902(missing-readme), ] Missing ./README.rst file. Template here: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
************* Module purchase_exception
purchase_exception/readme/USAGE.rst:1: [W7908(missing-newline-extrafiles), ]  Missing newline
```

1st error of missing README.rst file, I think it is not critical, right? As it will be created by Bot after merged (do I understand correctly?)

2nd error is from other module purchase_exception. When this happen, can I fix it in this PR? @pedrobaeza

File: purchase_deposit/__manifest__.py
Comment (nikul-serpentcs) [inline purchase_deposit/__manifest__.py]: @kittiu `12.0.1.0.0`

**Review by nikul-serpentcs — COMMENTED** [review]
Minor Change

File: purchase_deposit/__manifest__.py
Comment (kittiu) [reply purchase_deposit/__manifest__.py]: This is new module not migration, do I we need .1?

Comment (kittiu) [reply purchase_deposit/__manifest__.py]: Yes, it need :p. Fixed.

File: purchase_deposit/wizard/purchase_make_invoice_advance.py
Comment (nikul-serpentcs) [inline purchase_deposit/wizard/purchase_make_invoice_advance.py:52]: Remove One Line

**Review by nikul-serpentcs — COMMENTED** [review]
@kittiu Minor Change Because `Mr.Travis` Need more Love :heart:

File: purchase_deposit/wizard/purchase_make_invoice_advance.py
Comment (kittiu) [reply purchase_deposit/wizard/purchase_make_invoice_advance.py:52]: @nikul-serpentcs Thanks!, fixed.

**Comment by kittiu** [comment]
@OCA/procurement-purchase-maintainers can this be merged :) ?

File: purchase_deposit/models/purchase.py
Comment (HviorForgeFlow) [inline purchase_deposit/models/purchase.py:26]: typo: purhcase

File: purchase_deposit/wizard/purchase_make_invoice_advance.py
Comment (HviorForgeFlow) [inline purchase_deposit/wizard/purchase_make_invoice_advance.py]: Which is the usage of this field. I can figure out that could be used to create a deposit for multiple PO's, but I can't reproduce it.

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Code & Functional review

File: purchase_deposit/wizard/purchase_make_invoice_advance.py
Comment (kittiu) [reply purchase_deposit/wizard/purchase_make_invoice_advance.py]: Yes, I wasn't needed. This part try to be similar to SO, but you are right, there is no such action in PO. So, I removed it.
Thank you @hveficent

**Comment by kittiu** [comment]
@nikul-serpentcs all fixed. Would you help approve this? Thx!

**Review by HviorForgeFlow — APPROVED** [review]

**Review by ps-tubtim — APPROVED** [review]

**Pull request merged**
